db/post_migrate/20230222145758_backfill_reactions_count.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class BackfillReactionsCount < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  class StatusReaction < ApplicationRecord; end
+
+  def up
+    safety_assured do
+      StatusReaction.unscoped.in_batches do |reactions|
+        execute <<-SQL.squish
+          INSERT INTO status_stats (status_id, reactions_count, updated_at, created_at)
+          SELECT DISTINCT ON (status_id) status_id, count(*), MIN(created_at) AS created_at, MAX(updated_at) AS updated_at
+          FROM status_reactions WHERE status_id IN (#{reactions.map(&:status_id).uniq.join(', ')}) GROUP BY status_id
+          ON CONFLICT (status_id) DO UPDATE
+          SET reactions_count = EXCLUDED.reactions_count, updated_at = greatest(status_stats.updated_at, EXCLUDED.updated_at), created_at = least(status_stats.created_at, EXCLUDED.created_at)
```

Comment:
greatest and least don't seem to work as expected here, which makes me wonder if I should just drop that, even though that would mean the dates are not exactly correct.

Reply:
Apparently this was just caused by mixing the order in `insert into` and `select`?